src/playlist-maker.js

```diff
@@ -198,7 +199,11 @@ export default function factory(player, initialList, initialIndex = 0) {
     }
 
     if (Array.isArray(newList)) {
-
+     // The plugin should error out when rceiving a list with non-object values including null.
+      if (newList.filter(item => !!item && typeof item === 'object').length !== newList.length) {
+        player.off('loadstart');
+        throw new Error('playlist() received a list containing one or more non-object values');
```

Comment:
This is a breaking change.
Rather than erroring out, we should transform anything provided into a list of objects.

Reply:
The tests would need to be updated but only the assertions, I believe. Though, if we wanted, we can always keep the original value and return that outside the internals of the plugin.

Reply:
Actually, I think we do need to do something like that because it's possible that users are expecting the current behavior in the tests. While internally, we can do whatever we want.

Reply:
I agree.

Reply:
A potential way of doing it is in the transformed object add a `originalValue` property and then wherever we return the item, we return originalValue if it exists. But I can imagine there are other ways too.